Source/Core/VideoCommon/OnScreenUI.cpp

```diff
@@ -326,13 +327,83 @@ void OnScreenUI::DrawDebugText()
     ImGui::TextUnformatted(profile_output.c_str());
 }
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+void OnScreenUI::DrawChallenges()
+{
+  const std::map<std::string, std::unique_ptr<OSD::Icon>>* challenge_icons;
+  {
+    std::lock_guard lg{*AchievementManager::GetInstance()->GetLock()};
+    challenge_icons = AchievementManager::GetInstance()->GetChallengeIcons();
+  }
+
+  const std::string window_name = "challenges";
+
+  u32 icons_height = 0;
+  u32 icons_width = 0;
+  for (auto& entry : *challenge_icons)
+  {
+    // These *should* all be the same square size but you never know.
+    if (entry.second->height > icons_height)
+      icons_height = entry.second->height;
+    icons_width += entry.second->width;
+  }
+  ImGui::SetNextWindowPos(
+      ImVec2(ImGui::GetIO().DisplaySize.x - 20.f * m_backbuffer_scale - icons_height,
+             ImGui::GetIO().DisplaySize.y - 20.f * m_backbuffer_scale - icons_width));
+  ImGui::SetNextWindowSize(ImVec2(0.0f, 0.0f));
+  if (ImGui::Begin(window_name.c_str(), nullptr,
+                   ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoInputs |
+                       ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoSavedSettings |
+                       ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoNav |
+                       ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoFocusOnAppearing))
+  {
+    for (auto& entry : *challenge_icons)
+    {
+      auto texture_itr = m_challenge_texture_map.find(entry.first);
+      if (texture_itr == m_challenge_texture_map.end())
```

Comment:
```suggestion
      if (m_challenge_texture_map.find(entry.first) != m_challenge_texture_map.end())
        continue;
```
Continuing early lets us reduce the amount of indentation below.